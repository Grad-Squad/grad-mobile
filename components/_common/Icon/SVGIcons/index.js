import SVGIcons from './SVGIcons';

export default SVGIcons;
