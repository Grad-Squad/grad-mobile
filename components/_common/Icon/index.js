export { default as Icon } from './Icon';
export { default as PressableIcon } from './PressableIcon';
export { default as MaterialTypeIconsMap } from './MaterialTypeIconsMap';
