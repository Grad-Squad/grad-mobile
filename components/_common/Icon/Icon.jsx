import React from 'react';
import PropTypes from 'prop-types';
import { Colors } from 'styles';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import Feather from 'react-native-vector-icons/Feather';
import AntDesign from 'react-native-vector-icons/AntDesign';
import { TextPropType } from 'proptypes';
import SVGIcons from './SVGIcons';

const map = {
  question: AntDesign,
  close: Ionicons,
  'home-outline': Ionicons,
  'radio-button-checked': MaterialIcons,
  'radio-button-off': MaterialIcons,
  'bell-outline': MaterialCommunityIcons,
  search: Octicons,
  'add-circle-outline': MaterialIcons,
  bookmark: Feather,
  menu: Feather,
  'cards-outline': MaterialCommunityIcons,
  'format-list-checkbox': MaterialCommunityIcons,
  pdffile1: AntDesign,
  'image-outline': MaterialCommunityIcons,
  'ondemand-video': MaterialIcons,
  'dots-horizontal': MaterialCommunityIcons,
  'dots-vertical': MaterialCommunityIcons,
  'edit-2': Feather,
  'check-square': Feather,
  square: Feather,
  AddImage: SVGIcons,
  'delete-outline': MaterialIcons,
  'send':MaterialIcons,
  "arrow-left":MaterialCommunityIcons,
  'plus':Feather,
  'comment-outline':MaterialCommunityIcons,
};

const Icon = ({ name, size, color, style }) => {
  const IconLibrary = map[name] || map.question;
  return <IconLibrary name={name} size={size} color={color} style={style} />;
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string,
  style: TextPropType,
};
Icon.defaultProps = {
  size: 26,
  color: Colors.offBlack,
  style: {},
};

export default React.memo(Icon);
