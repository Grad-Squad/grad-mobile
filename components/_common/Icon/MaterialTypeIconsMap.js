import { IconNames } from "./Icon";

export default {
  Flashcards: IconNames.cards,
  MCQ: IconNames.checklist,
  PDF: IconNames.pdf,
  Images: IconNames.image,
  Video: IconNames.video,
};
