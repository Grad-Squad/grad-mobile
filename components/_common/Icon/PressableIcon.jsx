import React from 'react';
import PropTypes from 'prop-types';
import { Pressable } from 'react-native';
import pressableAndroidRipple from 'common/pressableAndroidRipple';
import { stylePropType } from 'proptypes';
import Icon from './Icon';

const PressableIcon = ({ onPress, style, pressableProps, ...rest }) => (
  <Pressable
    onPress={onPress}
    android_ripple={pressableAndroidRipple}
    style={style}
    // eslint-disable-next-line react/jsx-props-no-spreading
    {...pressableProps}
  >
    <Icon {...rest} />
  </Pressable>
);

PressableIcon.propTypes = {
  onPress: PropTypes.func.isRequired,
  style: stylePropType,
  // eslint-disable-next-line react/forbid-prop-types
  pressableProps: PropTypes.object,
};
PressableIcon.defaultProps = {
  style: {},
  pressableProps: {},
};

export default PressableIcon;
