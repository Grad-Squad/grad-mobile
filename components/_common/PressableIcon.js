import React from 'react';
import PropTypes from 'prop-types';
import { Pressable } from 'react-native';
import Icon from './Icon';
import pressableAndroidRipple from './pressableAndroidRipple';

const PressableIcon = ({ onPress, ...rest }) => (
  <Pressable onPress={onPress} android_ripple={pressableAndroidRipple}>
    <Icon {...rest} />
  </Pressable>
);

PressableIcon.propTypes = { onPress: PropTypes.func.isRequired };
PressableIcon.defaultProps = {};

export default PressableIcon;
