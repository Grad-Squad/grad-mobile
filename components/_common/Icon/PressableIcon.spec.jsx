import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import PressableIcon from './PressableIcon';
import { IconNames } from './Icon';

let pressableIcon;
const onPressMock = jest.fn();
beforeEach(() => {
  pressableIcon = render(<PressableIcon onPress={onPressMock} name={IconNames.question}/>);
  onPressMock.mockClear();
});

it('should match snapshot', () => {
  expect(pressableIcon.toJSON()).toMatchSnapshot();
});

it('should should call onPress when pressed', () => {
  fireEvent.press(pressableIcon.container);
  expect(onPressMock).toHaveBeenCalledTimes(1);
});
