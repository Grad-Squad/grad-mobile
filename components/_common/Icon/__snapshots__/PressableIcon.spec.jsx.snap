// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <Text />
</View>
`;
